crates/next-core/src/router.rs

```diff
@@ -154,16 +206,20 @@ fn route_executor(context: AssetContextVc, project_path: FileSystemPathVc) -> As
 pub async fn route(
     execution_context: ExecutionContextVc,
     request: RouterRequestVc,
+    next_config: NextConfigVc,
 ) -> Result<RouterResultVc> {
     let ExecutionContext {
         project_root,
         intermediate_output_path,
     } = *execution_context.await?;
     let project_path = wrap_with_next_js_fs(project_root);
     let context = node_evaluate_asset_context(Some(get_next_build_import_map(project_path)));
-    let router_asset = route_executor(context, project_path);
-    // TODO this is a hack to get these files watched.
-    let extra_configs = extra_configs(context, project_path);
+
+    let configs = config_assets(context, project_path, next_config.page_extensions());
+    let router_asset = route_executor(context, project_path, configs);
+
+    // TODO this is a hack to get this files watched.
```

Comment:
```suggestion
    // TODO this is a hack to get these files watched.
```

Reply:
Best would be to either bundle that or at least trace that. Same goes for the externals, we should trace them. But not sure if this means we loose the performance benefit from making them external^^